### ISSUE TITLE
HTTP/1.1 301 Moved Permanently connection redirect within --docker-run times out

### ISSUE THREAD

**Issue by josephmate** [telepresenceio/telepresence#1220]
The problem is reproducible every time by asking for an HTTP GET, and the connection gives back an HTTP/1.1 301 Moved Permanently

```
telepresence --docker-run --rm centos:centos7   curl -i -v http://ec2.amazonaws.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1576532273-094672-1474

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 80 (#0)
*   Trying 54.239.29.8...
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to ec2.amazonaws.com (54.239.29.8) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: ec2.amazonaws.com
> Accept: */*
>
  0     0    0     0    0     0      0      0 --:--:--  0:07:30 --:--:--     0* Empty reply from server
  0     0    0     0    0     0      0      0 --:--:--  0:07:30 --:--:--     0
* Connection #0 to host ec2.amazonaws.com left intact
curl: (52) Empty reply from server
T: Your process exited with return code 52.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1576532273-094672-1474
```

Summary:

telepresence --swap-deployment my-component --docker-run my-image results in my java application throwing `javax.net.ssl.SSLException: Read timed out` - I think the connection stalls, and never receives the HTTP 301 Moved Permanently because:
- `kubectl exec -t <podId> -- curl -i https://ec2.amazonaws.com` works on the original remote pod before running telepresence [1]
- `telepresence --docker-run --rm centos:centos7   curl -i https://ec2.amazonaws.com` times out [2]
- `telepresence --docker-run --rm centos:centos7   curl -i http://ec2.amazonaws.com` times out [6]
- `telepresence --docker-run --rm centos:centos7   curl -i https://www.google.com` works[3]
- `telepresence --docker-run --rm centos:centos7   ping -c 2 ec2.amazonaws.com` works [4]
- `docker run --rm centos:centos7   curl --insecure https://ec2.amazonaws.com` works [5]

How can I continue to debug this? Is there an argument I can provide to telepresence to work around this?

Long story:

```
telepresence --version
0.103
```

[1] My application run fine without telepresence. It's able to communicate with the ec2 api. Easiest way I can show that is will curl:
```
kubectl exec -t <podId> -- curl -i https://ec2.amazonaws.com 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 301 Moved Permanently
Location: https://aws.amazon.com/ec2
Content-Type: text/xml;charset=UTF-8
Content-Length: 0
Date: Mon, 16 Dec 2019 19:47:24 GMT
Server: AmazonEC2
```

When I run it within telepresence with --docker-run my application encounters:
```
 javax.net.ssl.SSLException: Read timed out 
```

[2] I think a GET to ec2.amazonaws.com reproduces the problem my java application faces:
```
telepresence --docker-run --rm centos:centos7   curl -i -v https://ec2.amazonaws.com
T: How Telepresence uses sudo: https://www.telepresence.io/reference/install#dependencies
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1576531775-01141-544

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 443 (#0)
*   Trying 54.239.28.168...
* Connected to ec2.amazonaws.com (54.239.28.168) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:--  0:04:59 --:--:--     0* Operation timed out after 300906 milliseconds with 0 out of 0 bytes received
  0     0    0     0    0     0      0      0 --:--:--  0:05:00 --:--:--     0
* Closing connection 0
curl: (28) Operation timed out after 300906 milliseconds with 0 out of 0 bytes received
T: Your process exited with return code 28.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1576531775-01141-544
```

[3] Notice that curling another url is fine:
```
telepresence --docker-run --rm centos:centos7   curl -i https://www.google.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1576522183-934478-94766

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/1.1 200 OK
Date: Mon, 16 Dec 2019 19:23:44 GMT
Expires: -1
.
.  output truncated
.
```

[4] Ping works:
```
telepresence --docker-run --rm centos:centos7   ping -c 2 ec2.amazonaws.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1576521869-8698041-76652

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
PING ec2.amazonaws.com (54.239.29.8) 56(84) bytes of data.
64 bytes from 54.239.29.8 (54.239.29.8): icmp_seq=1 ttl=37 time=47.1 ms
64 bytes from 54.239.29.8 (54.239.29.8): icmp_seq=2 ttl=37 time=44.7 ms

--- ec2.amazonaws.com ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 44.784/45.961/47.139/1.196 ms
T: Your process has exited.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1576521869-8698041-76652
```

[5] Running the curl with raw docker returns with no delay:
```
docker run --rm centos:centos7   curl -i -v  https://ec2.amazonaws.com
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 443 (#0)
*   Trying 54.239.28.168...
* Connected to ec2.amazonaws.com (54.239.28.168) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* SSL connection using TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
* Server certificate:
*       subject: CN=ec2.us-east-1.amazonaws.com
*       start date: Aug 12 00:00:00 2019 GMT
*       expire date: Aug 02 12:00:00 2020 GMT
*       common name: ec2.us-east-1.amazonaws.com
*       issuer: CN=Amazon,OU=Server CA 1B,O=Amazon,C=US
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: ec2.amazonaws.com
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Location: https://aws.amazon.com/ec2
< Content-Type: text/xml;charset=UTF-8
< Content-Length: 0
< Date: Mon, 16 Dec 2019 21:40:57 GMT
< Server: AmazonEC2
< 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host ec2.amazonaws.com left intact
HTTP/1.1 301 Moved Permanently
Location: https://aws.amazon.com/ec2
Content-Type: text/xml;charset=UTF-8
Content-Length: 0
Date: Mon, 16 Dec 2019 21:40:57 GMT
Server: AmazonEC2
```

[6] Running curl against http faces a similar issue to [2]:
```
telepresence --docker-run --rm centos:centos7   curl -i -v http://ec2.amazonaws.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1576532273-094672-1474

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 80 (#0)
*   Trying 54.239.29.8...
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to ec2.amazonaws.com (54.239.29.8) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: ec2.amazonaws.com
> Accept: */*
>
  0     0    0     0    0     0      0      0 --:--:--  0:07:30 --:--:--     0* Empty reply from server
  0     0    0     0    0     0      0      0 --:--:--  0:07:30 --:--:--     0
* Connection #0 to host ec2.amazonaws.com left intact
curl: (52) Empty reply from server
T: Your process exited with return code 52.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1576532273-094672-1474
```

**Comment by josephmate** [comment]
Anyone have points on how I can debug this? My guess is that the HTTP redirect causes a stall somewhere.

**Comment by aviau** [comment]
@josephmate Any updates on this? We just spen a few hours debbuging this without clues

**Comment by josephmate** [comment]
@aviau 
> Any updates on this?

Unfortunately, I gave up trying to use telepresence with the container that had a redirect.

> My guess is that is has to do with an incompatibility with telepresence's proxying methods and Amazon's HTTP server implementation?!?!?

It's not an Amazon's issue because you can reproduce it with other locations that have 301 moved permanently.

For example this curl works no problem in docker with google:
```
docker run --rm centos:centos7   curl -i -v  http://google.com
docker run --rm centos:centos7   curl -i -v  http://google.com
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to google.com port 80 (#0)
*   Trying 172.217.1.14...
* Connected to google.com (172.217.1.14) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: google.com
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Location: http://www.google.com/
< Content-Type: text/html; charset=UTF-8
< Date: Wed, 20 May 2020 01:38:40 GMT
< Expires: Fri, 19 Jun 2020 01:38:40 GMT
< Cache-Control: public, max-age=2592000
< Server: gws
< Content-Length: 219
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
< 
{ [data not shown]
100   219  100   219    0     0   3785      0 --:--:-- --:--:-- --:--:--  3842
* Connection #0 to host google.com left intact
HTTP/1.1 301 Moved Permanently
Location: http://www.google.com/
Content-Type: text/html; charset=UTF-8
Date: Wed, 20 May 2020 01:38:40 GMT
Expires: Fri, 19 Jun 2020 01:38:40 GMT
Cache-Control: public, max-age=2592000
Server: gws
Content-Length: 219
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN

<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>

```

but stalls again with telepresence:
```
telepresence --docker-run --rm centos:centos7   curl -i -v http://google.com
# Ctrl+C to exit after 1 minute
```

**Comment by isra17** [comment]
Hey @josephmate , @aviau coworker here. This is weird, in our case, we can't get anything from amazon.com, but google.com works just fine. The issue seems to happen behind the sshuttle proxy (Our container or `telepresence-local`), but not after (Say, in `telepresence-k8s` pod). After dumping the TCP in `telepresence-k8s` and do a basic inspection it seems we never get the data back from aws:

### Curl hang and fail

```
    1 0.000000000  172.17.0.12 → 176.32.98.166 TCP 74 52140 → 80 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=1974590672 TSecr=0 WS=128
    2 0.030236950 176.32.98.166 → 172.17.0.12  TCP 66 80 → 52140 [SYN, ACK] Seq=0 Ack=1 Win=8190 Len=0 MSS=1460 WS=64 SACK_PERM=1
    3 0.030302860  172.17.0.12 → 176.32.98.166 TCP 54 52140 → 80 [ACK] Seq=1 Ack=1 Win=64256 Len=0
    4 2.521341997  172.17.0.12 → 176.32.98.166 TCP 60 [TCP segment of a reassembled PDU]
    5 2.565424957 176.32.98.166 → 172.17.0.12  TCP 54 80 → 52140 [ACK] Seq=1 Ack=7 Win=2097088 Len=0
    <hang...>
    6 13.731062003 176.32.98.166 → 172.17.0.12  TCP 54 80 → 52140 [FIN, ACK] Seq=1 Ack=7 Win=2097088 Len=0
    7 13.732118285  172.17.0.12 → 176.32.98.166 TCP 54 52140 → 80 [ACK] Seq=7 Ack=2 Win=64256 Len=0
    8 13.738201947  172.17.0.12 → 176.32.98.166 TCP 54 52140 → 80 [FIN, ACK] Seq=7 Ack=2 Win=64256 Len=0
```

### Curl that work just fine

```
    1 0.000000000  172.17.0.12 → 176.32.98.166 TCP 74 42669 → 80 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=1974503295 TSecr=0 WS=128
    2 0.034215567 176.32.98.166 → 172.17.0.12  TCP 66 80 → 42669 [SYN, ACK] Seq=0 Ack=1 Win=8190 Len=0 MSS=1460 WS=64 SACK_PERM=1
    3 0.034284818  172.17.0.12 → 176.32.98.166 TCP 54 42669 → 80 [ACK] Seq=1 Ack=1 Win=64256 Len=0
    4 3.104623804  172.17.0.12 → 176.32.98.166 TCP 60 [TCP segment of a reassembled PDU]
    5 3.129768037 176.32.98.166 → 172.17.0.12  TCP 54 80 → 42669 [ACK] Seq=1 Ack=7 Win=2097088 Len=0
    6 3.129790921 176.32.98.166 → 172.17.0.12  TCP 54 [TCP Dup ACK 5#1] 80 → 42669 [ACK] Seq=1 Ack=7 Win=2097088 Len=0
    7 3.129812706 176.32.98.166 → 172.17.0.12  TCP 233 80 → 42669 [PSH, ACK] Seq=1 Ack=7 Win=2097088 Len=179
    8 3.129864670  172.17.0.12 → 176.32.98.166 TCP 54 42669 → 80 [ACK] Seq=7 Ack=180 Win=64128 Len=0
    9 3.129829894 176.32.98.166 → 172.17.0.12  TCP 54 80 → 42669 [FIN, ACK] Seq=180 Ack=7 Win=2097088 Len=0
```

My theory is that some cloud provider has weird packet router or restriction incompatible with the stack in use. We finally were able to fix the issue by two means:

  1. Use an Http Proxy inside our kubernetes cluster and use it to route our amazone request. (You can usually simply set `HTTP_PROXY=http-proxy:8080` or something similar from your telepresence container)

 2. We tried restricting the CIDR used by sshuttle to the actual kubernetes services CIDR (See: https://github.com/telepresenceio/telepresence/blob/master/telepresence/outbound/container.py#L146). We replaced `0/0` by the CIDR from `kubectl cluster-info dump | grep -m 1 service-cluster-ip-range` and everything work just fine afterward. We might open a PR soon if this is something the maintainers are interested about.

**Comment by josephmate** [comment]
> This is weird, in our case, we can't get anything from amazon.com, but google.com works just fine

@isra17 : I was wrong. @aviau is right. My kubernetes cluster died from neglect resulting in the hang. Google works for me too:

```
telepresence --docker-run --rm curlimages/curl    curl -i -v  http://google.com 
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1590002274-814173-35300

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 172.217.10.110:80...
* Connected to google.com (172.217.10.110) port 80 (#0)
> GET / HTTP/1.1
> Host: google.com
> User-Agent: curl/7.70.0-DEV
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Location: http://www.google.com/
< Content-Type: text/html; charset=UTF-8
< Date: Wed, 20 May 2020 19:18:09 GMT
< Expires: Fri, 19 Jun 2020 19:18:09 GMT
< Cache-Control: public, max-age=2592000
< Server: gws
< Content-Length: 219
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
< 
{ [219 bytes data]
100   219  100   219    0     0    931      0 --:--:-- --:--:-- --:--:--   931
* Connection #0 to host google.com left intact
HTTP/1.1 301 Moved Permanently
Location: http://www.google.com/
Content-Type: text/html; charset=UTF-8
Date: Wed, 20 May 2020 19:18:09 GMT
Expires: Fri, 19 Jun 2020 19:18:09 GMT
Cache-Control: public, max-age=2592000
Server: gws
Content-Length: 219
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN

<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
T: Your process has exited.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1590002274-814173-35300

T: Telepresence 0.105 is available (you're running 0.103). https://www.telepresence.io/reference/changelog
```

What's also strange is that I'm not able to reproduce the amazon issue consistently with curl anymore. When I filed the bug, curl stalled every time.

Example where connection stalls:
```
telepresence --docker-run --rm centos:centos7   curl -i -v https://ec2.amazonaws.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1590003022-235892-35904

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 443 (#0)
*   Trying 54.239.28.168...
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to ec2.amazonaws.com (54.239.28.168) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
  0     0    0     0    0     0      0      0 --:--:--  0:02:16 --:--:--     0
```
Note the output 2:16 is how long I waited before ctrl+c.

Example with no issue:
```
telepresence --docker-run --rm centos:centos7   curl -i -v https://ec2.amazonaws.com
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.
T: Starting network proxy to cluster using new Deployment telepresence-1590002895-945832-35821

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Setup complete. Launching your container.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* About to connect() to ec2.amazonaws.com port 443 (#0)
*   Trying 52.46.142.40...
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to ec2.amazonaws.com (52.46.142.40) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* SSL connection using TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
* Server certificate:
*       subject: CN=ec2.us-east-1.amazonaws.com
*       start date: May 06 00:00:00 2020 GMT
*       expire date: Apr 25 12:00:00 2021 GMT
*       common name: ec2.us-east-1.amazonaws.com
*       issuer: CN=Amazon,OU=Server CA 1B,O=Amazon,C=US
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: ec2.amazonaws.com
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Location: https://aws.amazon.com/ec2
< x-amzn-requestid: a7aee3e1-218e-4b4f-be11-d82b358eb05d
< Content-Type: text/xml;charset=UTF-8
< Content-Length: 0
< Date: Wed, 20 May 2020 19:28:34 GMT
< Server: AmazonEC2
< 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host ec2.amazonaws.com left intact
HTTP/1.1 301 Moved Permanently
Location: https://aws.amazon.com/ec2
x-amzn-requestid: a7aee3e1-218e-4b4f-be11-d82b358eb05d
Content-Type: text/xml;charset=UTF-8
Content-Length: 0
Date: Wed, 20 May 2020 19:28:34 GMT
Server: AmazonEC2

T: Your process has exited.
T: Exit cleanup in progress
T: Cleaning up Deployment telepresence-1590002895-945832-35821

T: Telepresence 0.105 is available (you're running 0.103). https://www.telepresence.io/reference/changelog
```

> We tried restricting the CIDR used by sshuttle to the actual kubernetes services CIDR (See: https://github.com/telepresenceio/telepresence/blob/master/telepresence/outbound/container.py#L146). We replaced 0/0 by the CIDR from kubectl cluster-info dump | grep -m 1 service-cluster-ip-range and everything work just fine afterward. We might open a PR soon if this is something the maintainers are interested about.

Thank you for sharing your solution. If I ever need to use use that container that talks to the ec2 api again, I will try out Option 2.

**Comment by donnyyung** [comment]
I believe this is no longer an issue in Telepresence 2, since we have improved cidr guessing, no longer use sshuttle. Here are the docs on how to install Telepresence (https://www.telepresence.io/docs/latest/install/), please re-open if you still see this issue in our latest version!
